partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt

```diff
@@ -23,9 +28,9 @@ import org.partiql.value.PartiQLValueExperimental
  */
 internal object PlanTransform {
 
-    fun transform(node: PartiQLPlan, onProblem: ProblemCallback): org.partiql.plan.PartiQLPlan {
+    fun transform(node: PartiQLPlan, missingOpBehavior: PartiQLPlanner.Session.MissingOpBehavior, onProblem: ProblemCallback, debugMode: Boolean): org.partiql.plan.PartiQLPlan {
```

Comment:
Again, let's get MissingOpBehavior and DebugMode into a single compiler flags construct. These shouldn't be passed independently like this. All compilation configuration should be contained in a single type

Reply:
https://github.com/partiql/partiql-lang-kotlin/pull/1385#discussion_r1523729759

I think MissingOpBehavior is a session mode, whereas the Debugging mode vs production mode should be a planner mode. 

>  let's get MissingOpBehavior and DebugMode into a single compiler flags construct. These shouldn't be passed independently like this. All compilation configuration should be contained in a single type

Could you please elaborate on why you think this is a better approach?